SoftDeletes/Core/DbContext.cs

```diff
@@ -77,11 +77,25 @@ protected virtual void SetNewEntitiesTimestamps()
         protected virtual void SetModifiedEntitiesTimestamps()
         {
             var updatedRecords = ChangeTracker.Entries()
-                .Where(x => x.State == EntityState.Modified && x.Entity is ITimestamps)
-                .Select(x => x.Entity as ITimestamps);
+                .Where(x => x.State == EntityState.Modified && (x.Entity is ITimestamps || x.Entity is ISoftDelete))
+                .Select(x => x.Entity);
             foreach (var record in updatedRecords) {
                 if (record == null) continue;
-                record.UpdatedAt = DateTime.Now;
+                
+                if(record is ISoftDelete)
+                {
+                    //TODO: Check how to remove SoftDelete attribute and how this might affect preventing reverting SoftDelete
```

Comment:
This might have impact if you are trying to set UpdatedAt to NULL. I will think how to handle that.